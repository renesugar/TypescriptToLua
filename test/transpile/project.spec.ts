import * as path from "path";
import { transpileProjectResult } from "./run";

test("should transpile", () => {
    const { diagnostics, emittedFiles } = transpileProjectResult(path.join(__dirname, "project", "tsconfig.json"));
    expect(diagnostics).not.toHaveDiagnostics();
    expect(emittedFiles).toMatchSnapshot();
});
