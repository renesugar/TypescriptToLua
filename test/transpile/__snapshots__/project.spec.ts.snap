// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should transpile 1`] = `
Array [
  Object {
    "name": "project/otherFile.lua",
    "text": "--[[ Generated with https://github.com/TypeScriptToLua/TypeScriptToLua ]]
local ____exports = {}
function ____exports.getNumber(self)
    return getAPIValue()
end
return ____exports
",
    "writeByteOrderMark": false,
  },
  Object {
    "name": "project/index.lua",
    "text": "--[[ Generated with https://github.com/TypeScriptToLua/TypeScriptToLua ]]
local ____exports = {}
local ____otherFile = require(\\"otherFile\\")
local getNumber = ____otherFile.getNumber
local myNumber = getNumber(nil)
setAPIValue(myNumber * 5)
return ____exports
",
    "writeByteOrderMark": false,
  },
]
`;
